compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -669,6 +697,64 @@ fn report_missing_placeholders(
     diag.emit();
 }
 
+fn report_redundant_placeholders(
```

Comment:
nit: documentation comment describing the addition this function adds would be nice